### DISCUSSION TITLE
Vimeo player not working in `crossOriginIsolated` page

### DISCUSSION THREAD

**Discussion by cduff** [vimeo/player.js#1021] [Bug reports]
### Expected Behavior

Vimeo player should work in a page that is [crossOriginIsolated](https://developer.mozilla.org/en-US/docs/Web/API/crossOriginIsolated).

### Actual Behavior

Vimeo player fails in a page served with HTTP response header [Cross-Origin-Embedder-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Embedder-Policy) values of `require-corp` or `credentialless`.

### Steps to Reproduce

- See reproduction project: https://stackblitz.com/edit/vimeo-cross-origin-isolated-issue?file=serve.json
- `npm start` and view output in own tab. *(Even the default page fails in the StackBlitz embedded preview due to cross origin isolation.)*
- `/index.html` works fine as there is no HTTP response `Cross-Origin-Embedder-Policy` header value set.
- `/require-corp` and `/credentialless` fail due to their respective HTTP response `Cross-Origin-Embedder-Policy` header values.

### Solution

Vimeo should be setting the [Cross-Origin-Resource-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Resource-Policy) response header value to `cross-origin` on HTTP responses as appropriate?

### My environment

Latest Chrome (121.0.6167.140) on latest Windows 11 Pro (23H2, 22631.3085).

**Comment by rkrishnan8594** [comment]
Hey @cduff - thanks for filing this issue. I'm able to reproduce using the StackBlitz test app that you put together. I've got this issue captured in an internal ticket and I'm hoping that we can address it within the next couple of weeks. I'll follow-up on this issue when we have an update.

**Comment by cduff** [comment]
Hi @bdougherty, can you let us know why you converted this from an issue into a discussion?

@rkrishnan8594 any update on your internal ticket?

**Comment by cduff** [comment]
Any update on this @bdougherty, @rkrishnan8594? I tested this today and can see that it is still an issue.

**Comment by rkrishnan8594** [comment]
Hi @cduff - the reason we moved this to a discussion is because it's related to the Vimeo player itself, rather than the Player.js library. Regardless, apologies for the delay on a resolution for this issue. My hope is that we can deploy a fix within the month of July.

**Reply by davidfstr** [reply]
This issue still persists in Jan 2025. 

I can workaround a lack of Cross-Origin-Resource-Policy = cross-origin header on recent versions of Chrome using the experimental [IFrame credentialless](https://developer.mozilla.org/en-US/docs/Web/Security/IFrame_credentialless) feature applied to the IFRAME tag used to load Vimeo. However that feature doesn't work in most (any?) other browsers right now, at least not in Firefox or Safari.

**Reply by oom-** [reply]
@rkrishnan8594
> the month of July.

Maybe you was speaking of July 2025 ? 😶

My website has: 
```
cross-origin-embedder-policy: credentialless
cross-origin-opener-policy: same-origin
cross-origin-resource-policy: same-origin
```

And it cannot load the player:
![image](https://github.com/user-attachments/assets/9bf67406-7fd3-4003-8613-a0a198b63db7)

Where do we need to rise this up? please add a CORP policy header `Cross-Origin-Resource-Policy: cross-origin` and it's done.

**Reply by rkrishnan8594** [reply]
@cduff @oom- @davidfstr

Thanks for your patience on this issue, apologies for the long delay between updates. I looked into this a bit more this weekend, and while I can add `Cross-Origin-Resource-Policy: cross-origin` header to the player's HTML response, it unfortunately might not resolve your issues. If your page has a COEP of `require-corp` or `credentialless` set, that means that any iframe embed also needs a COEP with one of those values set, which is not a change we're able to make right now. I'll add the CORP header regardless, but it might be best to add the `credentialless` attribute to the iframe if you're still experiencing issues.

**Reply by rkrishnan8594** [reply]
Update - this header is now being served on production, so I'm going to close this discussion.

**Reply by davidfstr** [reply]
Thanks @rkrishnan8594 ! <s>I've verified that the `credentialless` workaround is no longer needed.</s>

Ah. It looks like I still would need the `Cross-Origin-Embedder-Policy: cross-origin` to remove the `credentialless` workaround fully. And I see that you can't add that right now:

> If your page has a COEP of require-corp or credentialless set, that means that any iframe embed also needs a COEP with one of those values set, which is not a change we're able to make right now.

**Reply by davidfstr** [reply]
@rkrishnan8594 Do you know of an existing discussion/issue tracking the addition of the `Cross-Origin-Embedder-Policy: cross-origin` header that I can subscribe to or vote on? (If not, I can create one.)

**Reply by rkrishnan8594** [reply]
There isn't an existing feature request for that, as far as I know. Feel free to create one. For context, this would be a tough change for us to make for the same reason you're reaching out - if we applied a `require-corp` COEP, we'd need to ensure that any non-CORS cross-origin requests we make have this header set as well, and if we applied a `credentialless` COEP, we'd lose access to credentials sent along with those requests, which are important in some cases.
